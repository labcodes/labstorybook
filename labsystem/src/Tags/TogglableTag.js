import React from "react";
import PropTypes from "prop-types";
import AbstractTag from "./AbstractTag";

export default class TogglableTag extends React.Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
  };

  onClick = () => {
    const { text } = this.props;
    // eslint-disable-next-line no-alert
    alert(text);
  };

  render() {
    const { text } = this.props;
    return (
      <AbstractTag
        togglable
        text={text}
        onClick={this.onClick}
        {...this.props}
      />
    );
  }
}
