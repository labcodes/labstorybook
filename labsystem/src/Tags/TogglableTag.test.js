import React from "react";
import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";

import TogglableTag from "./TogglableTag";

describe("TogglableTag", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<TogglableTag text="Test TogglableTag" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders as expected with disabled as true", async () => {
    const renderedComponent = renderer
      .create(<TogglableTag text="Test nount disabled TogglableTag" disabled />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const mountedComponent = mount(
      <TogglableTag text="Test mount disabled TogglableTag" disabled />
    );
    expect(mountedComponent.find("AbstractTag").prop("disabled")).toEqual(true);
  });

  it("renders as expected with outline as true", async () => {
    const renderedComponent = renderer
      .create(<TogglableTag text="Test render outline TogglableTag" outline />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const mountedComponent = mount(
      <TogglableTag text="Test mount outline TogglableTag" outline />
    );
    expect(mountedComponent.find("AbstractTag").prop("outline")).toEqual(true);
  });

  it("renders as expected with a yellow color", async () => {
    const renderedComponent = renderer
      .create(
        <TogglableTag text="Test render yellow TogglableTag" color="yellow" />
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const mountedComponent = mount(
      <TogglableTag text="Test mount yellow TogglableTag" color="yellow" />
    );
    expect(mountedComponent.find("AbstractTag").prop("color")).toEqual("yellow");
  });

  it("renders with a pale skin if not isOn", async () => {
    const renderedComponent = renderer
      .create(
        <TogglableTag text="Test render pale TogglableTag" isOn={false} />
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const mountedComponent = mount(
      <TogglableTag text="Test mount pale TogglableTag" isOn={false} />
    );
    expect(mountedComponent.find("AbstractTag").prop("skin")).toEqual("pale");
  });

  it("renders with a vivid skin and a checked icon if isOn", async () => {
    const renderedComponent = renderer
      .create(<TogglableTag text="Test render isOn TogglableTag" isOn />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const mountedComponent = mount(
      <TogglableTag text="Test mount isOn TogglableTag" isOn />
    );
    expect(mountedComponent.find("AbstractTag").prop("skin")).toEqual("vivid");
    expect(mountedComponent.find("AbstractTag").prop("icon")).toEqual("check");
  });

  it("calls prop.onClick when clicked", async () => {
    const mockOnClick = jest.fn();
    const shallowedTogglableTag = shallow(
      <TogglableTag text="Test click on TogglableTag" onClick={mockOnClick} />
    );
    expect(mockOnClick.mock.calls.length).toEqual(0);
    shallowedTogglableTag.simulate("click");
    expect(mockOnClick.mock.calls.length).toEqual(1);
  });
});
