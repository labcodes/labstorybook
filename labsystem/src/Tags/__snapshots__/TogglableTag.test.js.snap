// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TogglableTag renders as expected with a yellow color 1`] = `
<span
  className="lab-tag lab-tag--togglable lab-tag--yellow-pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  Test render yellow TogglableTag
</span>
`;

exports[`TogglableTag renders as expected with disabled as true 1`] = `
<span
  className="lab-tag lab-tag--togglable lab-tag--disabled lab-tag--pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  Test nount disabled TogglableTag
</span>
`;

exports[`TogglableTag renders as expected with outline as true 1`] = `
<span
  className="lab-tag lab-tag--togglable lab-tag--outline lab-tag--pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  Test render outline TogglableTag
</span>
`;

exports[`TogglableTag renders with a pale skin if not isOn 1`] = `
<span
  className="lab-tag lab-tag--togglable lab-tag--pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  Test render pale TogglableTag
</span>
`;

exports[`TogglableTag renders with a vivid skin and a checked icon if isOn 1`] = `
<span
  className="lab-tag lab-tag--togglable lab-tag--vivid lab-tag--has-left-icon"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  <span
    className="lab-icon lab-icon--check lab-icon--black75 lab-icon--petit lab-icon--check-icon"
  />
  Test render isOn TogglableTag
</span>
`;

exports[`TogglableTag renders with base props 1`] = `
<span
  className="lab-tag lab-tag--togglable lab-tag--pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  Test render TogglableTag
</span>
`;
